src/index.js

```diff
@@ -29,7 +30,14 @@ async function run() {
 
     // install Python dependency definitions in user-defined requirements.txt file path
     console.log("[*] Installing Python dependencies...");
-    await exec.exec(`python -m pip install -r ${requirementsPath}`);
+    if (constraintsPath === "none") {
+      await exec.exec(`python -m pip install -r ${requirementsPath}`);
+    } else {
+      await exec.exec(
+        `python -m pip install -r ${requirementsPath} -c ${constraintsPath}`
```

Comment:
It's best to use an env var for this, since pip doesn't bypass constraints to internal ephemeral envs made for building sdists via PEP 517.
```suggestion
        `PIP_CONSTRAINT='${constraintsPath}' python -m pip install -r ${requirementsPath}`
```
(not sure if this syntax works in JS, you may need to figure out how to pass env vars in a different way...)